### PR TITLE
Remaining Tensor API fixes(CPU) - dims() -> sizes()

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary: Remaining fixes so that D12812029 will compile

Differential Revision: D13535559

### CODE REVIEW

**Comment by ezyang** [comment]
@jerryzh168 This diff seems to do more than just rename dims() to sizes()?

File: caffe2/mobile/contrib/ios/ios_caffe_predictor.cc [hunk 1]
```diff
@@ -50,8 +50,7 @@ Caffe2IOSPredictor::Caffe2IOSPredictor(const caffe2::NetDef& init_net,
 
 void Caffe2IOSPredictor::run(const Tensor& inData, Tensor& outData, std::string& errorMessage) {
   FLAGS_caffe2_force_shared_col_buffer = true;
-  caffe2::Tensor input(caffe2::CPU);
-  input.Resize(inData.dims);
+  caffe2::Tensor input(inData.dims, caffe2::CPU);
```
Comment (ezyang) [inline caffe2/mobile/contrib/ios/ios_caffe_predictor.cc]: Is there any particular reason we can't rewrite these in the factory function style, e.g.,

```
auto input = empty(inData.dims);
```

File: caffe2/mobile/contrib/ios/mpscnn/mpscnn.mm [hunk 2]
```diff
@@ -256,12 +256,12 @@ void computeOutputHW(
     int W,
     int* OH,
     int* OW) {
-  Tensor input(CPU), output(CPU);
-  input.Resize(1, 1, H, W);
+  std::vector<int64_t> dims = {1, 1, H, W};
```
Comment (ezyang) [inline caffe2/mobile/contrib/ios/mpscnn/mpscnn.mm]: I don't know if the constructor can handle initializer lists, but I know that empty definitely can:

```
Tensor input = empty({1, 1, H, W});
```

No need to allocate an entire vector here.

File: caffe2/queue/rebatching_queue.cc [hunk 3]
```diff
@@ -42,7 +42,7 @@ void concat(
       CAFFE_ENFORCE_EQ(inputZero[j].itemsize(), input.itemsize());
       CAFFE_ENFORCE_EQ(inputZero[j].ndim(), input.dim());
       for (int k = 0; k < input.dim(); ++k) {
-        CAFFE_ENFORCE_EQ(input.sizes()[k], inputZero[j].dims()[k]);
+        CAFFE_ENFORCE_EQ(input.sizes()[k], inputZero[j].size(k));
```
Comment (ezyang) [inline caffe2/queue/rebatching_queue.cc]: good

**Comment by ezyang** [comment]
Everything else looks ok

**Comment by jerryzh168** [comment]
> @jerryzh168 This diff seems to do more than just rename dims() to sizes()?

Yeah, but it's mostly renaming dims() to sizes()...

File: caffe2/mobile/contrib/ios/mpscnn/mpscnn.mm (hunk 2 above)
Comment (jerryzh168) [reply caffe2/mobile/contrib/ios/mpscnn/mpscnn.mm]: looks like Tensor here is just used for record sizes, no actual memory is allocated...

File: caffe2/mobile/contrib/ios/ios_caffe_predictor.cc (hunk 1 above)
Comment (jerryzh168) [reply caffe2/mobile/contrib/ios/ios_caffe_predictor.cc]: will change to caffe2::empty

File: caffe2/mobile/contrib/ios/ios_caffe_predictor.cc [hunk 4]
```diff
@@ -50,8 +50,7 @@ Caffe2IOSPredictor::Caffe2IOSPredictor(const caffe2::NetDef& init_net,
 
 void Caffe2IOSPredictor::run(const Tensor& inData, Tensor& outData, std::string& errorMessage) {
   FLAGS_caffe2_force_shared_col_buffer = true;
-  caffe2::Tensor input(caffe2::CPU);
-  input.Resize(inData.dims);
+  caffe2::Tensor input = caffe2::empty(inData.dims, at::dtype<uint8_t>().device(caffe2::CPU));
```
Comment (ezyang) [inline caffe2/mobile/contrib/ios/ios_caffe_predictor.cc]: hum, at some point we should standardize on using kCPU, probably (not for this PR)

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**